source/gfoidl.DataCompression/ICompression.cs

```diff
@@ -12,6 +12,6 @@ public interface ICompression
         /// </summary>
         /// <param name="data">Input data</param>
         /// <returns>The compressed / filtered data.</returns>
-        IEnumerable<DataPoint> Process(IEnumerable<DataPoint> data);
+        DataPointIterator Process(IEnumerable<DataPoint> data);
```

Comment:
Check if this is a breaking change.

Reply:
A quick check / test doesn't show any problems. But this has to be proven thoroughly.

If it is a breaking change it should go for 2.0.0

Reply:
| target | breaking change |  
| -- | -- |  
| .net Full | ✔️ |  
| .net Core | ✔️ |

I don't what I tested yesterday in the late evening :wink: -- it is definitely and as expected a breaking change (change of method signature). So this has to go for version 2.0.0